src/components/structures/MatrixChat.js

```diff
@@ -635,23 +657,35 @@ module.exports = React.createClass({
                 action: 'start_post_registration',
             });
         } else if (screen.indexOf('room/') == 0) {
-            var roomString = screen.split('/')[1];
+            var roomString = screen.substring(5);
+            var eventId;
+
+            // extract event id, if one is given
+            var idx = roomString.indexOf('/');
+            if (idx >= 0) {
+                eventId = roomString.substring(idx+1);
+                roomString = roomString.substring(0, idx);
```

Comment:
Presumably `/{eventId}` is tacked on the end of the URL. Might it be worth handling the case where there are multiple `/` characters (or rather, grabbing the last `/`) in case we sprout paths? At the very least, I think this would match your intent better.